--- proc.h	2016-09-24 17:49:22.884443288 +0530
+++ proc1.h	2016-09-24 17:24:31.843432956 +0530
@@ -10,7 +10,7 @@
   volatile uint started;       // Has the CPU started?
   int ncli;                    // Depth of pushcli nesting.
   int intena;                  // Were interrupts enabled before pushcli?
-  
+
   // Cpu-local storage variables; see below
   struct cpu *cpu;
   struct proc *proc;           // The currently-running process.
@@ -66,6 +66,7 @@
   struct file *ofile[NOFILE];  // Open files
   struct inode *cwd;           // Current directory
   char name[16];               // Process name (debugging)
+  int priority;                // Priority for the process
 };
 
 // Process memory is laid out contiguously, low addresses first:
